CMakeLists.txt

```diff
@@ -0,0 +1,208 @@
+# We need 3.12 or later, so that we can set policy CMP0074; see below.
+cmake_minimum_required(VERSION 3.12)
+
+# We want find_path() and find_library() to honor {packagename}_ROOT
+if(POLICY CMP0074)
+  cmake_policy(SET CMP0074 NEW)
+endif()
+
+set(PCAPPP_VERSION "22.05+")
+
+# MAIN_PROJECT CHECK
+set(PCAPPP_MAIN_PROJECT OFF)
+if(CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+  set(PCAPPP_MAIN_PROJECT ON)
+endif()
+
+project(
+  PcapPlusPlus
+  DESCRIPTION "PcapPlusPlus is a multiplatform C++ library for capturing, parsing and crafting of network packets."
+  HOMEPAGE_URL "https://pcapplusplus.github.io/")
+
+# Include our custom CMake modules
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules/")
+include(CMakeDependentOption)
+include(CMakePackageConfigHelpers)
+include(DetectCompiler)
+include(DetectOS)
+include(GNUInstallDirs)
+
+# Declare install folders location
+set(PCAPPP_INSTALL_BINDIR ${CMAKE_INSTALL_BINDIR})
+set(PCAPPP_INSTALL_CMAKEDIR ${CMAKE_INSTALL_LIBDIR}/cmake/pcapplusplus)
+set(PCAPPP_INSTALL_INCLUDEDIR ${CMAKE_INSTALL_INCLUDEDIR}/pcapplusplus)
+set(PCAPPP_INSTALL_LIBDIR ${CMAKE_INSTALL_LIBDIR})
+
+# Build options (Turn on Examples and Tests if it's the main project)
+option(PCAPPP_BUILD_EXAMPLES "Build Examples" ${PCAPPP_MAIN_PROJECT})
+cmake_dependent_option(
+  PCAPPP_BUILD_TUTORIALS
+  "Build Tutorials"
+  OFF
+  "PCAPPP_BUILD_EXAMPLES"
+  OFF)
+option(PCAPPP_BUILD_TESTS "Build Tests" ${PCAPPP_MAIN_PROJECT})
+option(BUILD_SHARED_LIBS "Build using shared libraries" OFF)
+
+option(PCAPPP_USE_DPDK "Setup PcapPlusPlus with DPDK. In this case you must also set DPDK_ROOT")
+option(PCAPPP_USE_PF_RING "Setup PcapPlusPlus with PF_RING. In this case you must also set PF_RING_ROOT")
+option(PCAPPP_INSTALL "Install Pcap++" ${PCAPPP_MAIN_PROJECT})
+
+# Set C++11
+set(CMAKE_CXX_STANDARD 11)
+# popen()/pclose() are not C++ standards
+set(CMAKE_CXX_EXTENSIONS ON)
+
+# Usually on Windows PCAP_ROOT and Packet_ROOT are at the same location
+if(PCAPPP_TARGET_PLATFORM_WINDOWS)
```

Comment:
For long term maintenance, wouldn't it be easier to use standard CMake variables such as `WIN32`?

Reply:
Correct but how do you deal with FreeBSD or Android?

Reply:
Correct, I will give it a try.

You seems to have more Cmake experience than me. Do you agree with the user Options and overall cmake architecture?

I'm quite new to cmake and all these internal variables doesn't bother me as they will stay internal and we can fix them later but I would be glad to hear your feedback on the user Options and the overall architecture to be sure it's properly design and we will not have to break them in the next MR.

Reply:
I have dropped the DetectOS.cmake, thanks!